src/helpers/modeRegistry.ts

```diff
@@ -31,6 +33,24 @@ class ModeRegistry {
         });
     }
 
+    static modesRegisterFromDir(dirPath: string): void {
+        const modes: any[] = [];
+
+        const files = fs.readdirSync(dirPath);
```

Comment:
Consider adding error handling for the file system operations. This could help in cases where the directory does not exist or there are issues with file permissions. [important]